fuzzing/uri_parser/main.c

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 2022 Bennet Blischke <bennet.blischke@haw-hamburg.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include <err.h>
+#include <unistd.h>
+
+#include "uri_parser.h"
+#include "fuzzing.h"
+
+int main(void)
+{
+    size_t input_len;
+    char *input_buf = (char *)fuzzing_read_bytes(STDIN_FILENO, &input_len);
+
+    if (input_buf == NULL) {
+        errx(EXIT_FAILURE, "fuzzing_read_bytes failed");
+    }
+
+    uri_parser_result_t uri_res;
+
+    uri_parser_process(&uri_res, input_buf, input_len);
+
+    exit(EXIT_SUCCESS);
```

Comment:
For the `uri_parser` you might also be able to just use `NATIVE_AUTO_EXIT` but explicitly calling exit is of cause also fine.

Reply:
Why would one prefer one over the other?